sklearnex/cluster/k_means.py

```diff
@@ -1,25 +1,367 @@
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-# ===============================================================================
-
-from daal4py.sklearn.cluster import KMeans
-from onedal._device_offload import support_usm_ndarray
-
-# Note: `sklearnex.cluster.KMeans` only has functional
-# sycl GPU support. No GPU device will be offloaded.
-KMeans.fit = support_usm_ndarray(queue_param=False)(KMeans.fit)
-KMeans.fit_predict = support_usm_ndarray(queue_param=False)(KMeans.fit_predict)
-KMeans.predict = support_usm_ndarray(queue_param=False)(KMeans.predict)
-KMeans.score = support_usm_ndarray(queue_param=False)(KMeans.score)
+# ==============================================================================
+# Copyright 2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import logging
+
+from daal4py.sklearn._utils import daal_check_version
+
+if daal_check_version((2023, "P", 200)):
+
+    import warnings
+
+    import numpy as np
+    from scipy.sparse import issparse
+    from sklearn.cluster import KMeans as sklearn_KMeans
+    from sklearn.utils._openmp_helpers import _openmp_effective_n_threads
+    from sklearn.utils.validation import (
+        _check_sample_weight,
+        _deprecate_positional_args,
+        _num_samples,
+        check_is_fitted,
+    )
+
+    from daal4py.sklearn._n_jobs_support import control_n_jobs
+    from daal4py.sklearn._utils import sklearn_check_version
+    from onedal.cluster import KMeans as onedal_KMeans
+    from onedal.utils import _is_csr
+
+    from .._device_offload import dispatch, wrap_output_data
+    from .._utils import PatchingConditionsChain
+
+    @control_n_jobs(decorated_methods=["fit", "predict", "transform", "fit_transform"])
+    class KMeans(sklearn_KMeans):
+        __doc__ = sklearn_KMeans.__doc__
+        n_iter_, inertia_ = None, None
+        labels_, cluster_centers_ = None, None
```

Comment:
Doesn't make sense add this into __init__ ?

Reply:
I think, they can be removed safely